src/wasm/wasm-binary.cpp

```diff
@@ -795,6 +795,33 @@ void WasmBinaryWriter::writeNames() {
   // TODO: label, type, and element names
   // see: https://github.com/WebAssembly/extended-name-section
 
+  // GC field names
+  if (wasm->features.hasGC()) {
+    std::vector<HeapType> namedTypes;
+    for (auto& type : types) {
+      if (type.isStruct() && wasm->typeNames.count(type) &&
+          !wasm->typeNames.at(type).fieldNames.empty()) {
+        namedTypes.push_back(type);
+      }
+    }
+    if (!namedTypes.empty()) {
+      auto substart =
+        startSubsection(BinaryConsts::UserSections::Subsection::NameField);
+      o << U32LEB(namedTypes.size());
+      for (Index i = 0; i < namedTypes.size(); i++) {
+        auto type = namedTypes[i];
+        o << U32LEB(i);
+        auto& fieldNames = wasm->typeNames.at(type).fieldNames;
```

Comment:
I would find this more readable with the type spelled out here instead of `auto`. That way it would be easier to figure out what is going on in the loop below. Just from the `fieldNames` variable name, I wouldn't have expected it to be a collection of pairs.

Reply:
Fair point, added.